src/api/ledger/transaction.js

```diff
@@ -62,11 +62,19 @@ function getTransactionAsync(identifier: string, options: TransactionOptions,
       error = new errors.NotFoundError('Transaction not found');
     }
 
+    const minLedgerVersion = options.minLedgerVersion;
+
     if (error instanceof errors.NotFoundError
-        && !utils.hasCompleteLedgerRange(remote,
-            options.minLedgerVersion, maxLedgerVersion)) {
-      callback(new errors.MissingLedgerHistoryError('Transaction not found,'
-        + ' but the server\'s ledger history is incomplete'));
+        && !utils.hasCompleteLedgerRange(remote, minLedgerVersion,
+                                         maxLedgerVersion)) {
+      if (minLedgerVersion && maxLedgerVersion
+          && utils.isNonContiguousLedgerRange(remote, minLedgerVersion, 
+                                         maxLedgerVersion)) {
+        callback(new errors.NonContiguousLedgerHistoryError());
+      } else if (maxLedgerVersion 
+                 && utils.isPendingLedgerVersion(remote, maxLedgerVersion)) {
+        callback(new errors.PendingLedgerError());
+      }
```

Comment:
The if/else doesn't cover all the cases. I think it's probably easiest to check for PendingLedgerVersionError (looks like typo above), then have an else (not else if) for NonContiguousLedgerHistoryError.